internal/controller/controller.go

```diff
@@ -14,111 +18,265 @@ import (
+			return
+		}
+		// if connected
+		res.WriteHeader(http.StatusOK)
+
 	}
+}
+
+func (c *Controller) CreateShortURLJSONBatchHandler(ctx context.Context) http.HandlerFunc {
+	return func(res http.ResponseWriter, req *http.Request) {
+		//check content-type (application/json)
+		if ok := strings.Contains(req.Header.Get("Content-Type"), "application/json"); !ok {
+			c.logger.Warn("[WARNING]", zap.String("body", "no content type application/json"), zap.String("method", req.Method), zap.String("url", req.URL.Path))
+			http.Error(res, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+
+		// check the body: TODO: del reuse bbody
+
+		//check the body
+		body, err := io.ReadAll(req.Body)
+		if err != nil || len(body) == 0 {
+			c.logger.Info("Body is empty!", zap.Error(err))
+			http.Error(res, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+
+		// Replace the body with a new reader after reading from the original
+		req.Body = io.NopCloser(bytes.NewBuffer(body))
+
+		// create the array of structures to deserialize data
+		var desBatchStruct []models.JSONBatStructToDesReq
+
+		//TODO: [MENTOR] Should I implement such a check?
+		/*
+			dec := json.NewDecoder(req.Body)
+				if err := dec.Decode(&request); err != nil {
+					httpError.RespondWithError(res, http.StatusInternalServerError, "Invalid JSON body")
+					return
+				}
+		*/
 
+		var buf bytes.Buffer
+		// feed data from the body into the buffer
+		if _, err := buf.ReadFrom(req.Body); err != nil {
+			c.logger.Error("[ERROR]", zap.Error(err))
+			http.Error(res, http.StatusText(http.StatusBadRequest), http.StatusBadRequest) // TODO [MENTOR]: BadRequest or InternalServerError?
+			return
+		}
+		defer req.Body.Close() // TODO [MENTOR] I didn't assign it should I close it??
+		// deserialize JSON batch into desBatchStruct
+		if err := json.Unmarshal(buf.Bytes(), &desBatchStruct); err != nil {
+			c.logger.Error("[ERROR]", zap.Error(err))
+			http.Error(res, err.Error(), http.StatusBadRequest) // TODO [MENTOR]: BadRequest or InternalServerError?
+			return
+		}
+
+		// create an auxiliary array of structures
+		idURLAl, err := service.CreateBatchIDOrSh(&desBatchStruct)
+		if err != nil {
+			c.logger.Error("[ERROR]", zap.Error(err))
+			http.Error(res, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError) // TODO [MENTOR]: BadRequest or InternalServerError?
+			return
+		}
+
+		// save data into the database and create respBatch for response
+		respBatch, err := c.storage.SaveBatchURL(ctx, idURLAl)
+		if err != nil {
+			c.logger.Error("[ERROR]", zap.Error(err))
+			http.Error(res, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+		// create the array of structs to add base url to response
+		response := make([]models.JSONBatStructToSerResp, 0) //TODO [MENTOR] is it a good allocation or I should use len?
```

Comment:
```suggestion
		response := make([]models.JSONBatStructToSerResp, 0, len(respBatch)) //TODO [MENTOR] is it a good allocation or I should use len?
```
Хороший, правильный вопрос! https://t.me/bushigo/12

Reply:
done